src/screens/Documents/View/index.js

```diff
@@ -112,42 +130,182 @@ class DocumentsViewScreen extends NavigatorComponent<Props & DocumentsState & Ac
+            style={styles.preview}
+            resizeMode='contain'
+          />
         </View>
         <View style={styles.metadataContainer}>
-          <Text style={styles.headline}>
-            {document.name}
-          </Text>
-          <Text style={styles.footnote}>
-            {document.description}
-          </Text>
+          <ScrollView>
+            <Text style={styles.headline}>{document.name}</Text>
+            <Text style={styles.footnote}>{document.description}</Text>
+            {!isUploading && (
+              <Button
+                enabled
+                style={styles.actionButton}
+                title={i18n
+                  .t('screens.documentView.submitdocument')
+                  .toUpperCase()}
+                onPress={this.onDocumentSubmit}
+                styleTitle={styles.settingsText}
+              />
+            )}
+            {isUploading && document.status && (
+              <DocumentListItem
+                id={1}
+                name={i18n.t('screens.documentView.status')}
+                value={document.status}
+                disclosureIconVisible={false}
+                onPress={() => console.log('hello')}
+              />
+            )}
+            {document.tx_hash &&
+              document.doc_hash &&
+              document.signature &&
+              documentDetail.map((index, id) => (
+                <DocumentListItem
+                  id={id}
+                  name={index.name}
+                  value={index.value}
+                  disclosureIconVisible={false}
+                  onPress={() => this.displayDocumentValues(index)}
+                />
+              ))}
+          </ScrollView>
         </View>
         <MoreMenuModal
           visible={this.state.moreMenuVisible === true}
           onCancel={() => this.setState({ moreMenuVisible: false })}
-          options={[{
-            text: i18n.t('screens.documentView.actions.edit'),
-            onPress: this.onSelectEdit,
-          }, {
-            text: i18n.t('screens.documentView.actions.delete'),
-            onPress: this.onSelectDelete,
-          }]
-          }
+          options={[
+            {
+              text: i18n.t('screens.documentView.actions.edit'),
+              onPress: this.onSelectEdit,
+            },
+            {
+              text: i18n.t('screens.documentView.actions.delete'),
+              onPress: this.onSelectDelete,
+            },
+          ]}
         />
+        <Modal isVisible={this.state.visibleTxModal}>
+          <View>
+            <TouchableHighlight
+              onPress={() => {
+                this.txModalClose();
+              }}
+            >
+              <Icon name='ios-close' style={styles.closeIcon} />
+            </TouchableHighlight>
+          </View>
+          {this.state.txhash &&
+            <WebView
+              source={{
+                      uri: `https://etherscan.io/address/${
```

Comment:
It's not correct to always show etherscan, since we can have testing accounts. We have the same code in viewing transactions, so it can be extracted to something like `function etherscanURL(networkType: NetworkType, hash: string): string`

Reply:
@seland Added a function for this